src/kitovu/sync/syncing.py

```diff
@@ -7,26 +7,34 @@
 import stevedore.driver
 
 from kitovu import utils
-from kitovu.sync import syncplugin
+from kitovu.sync.syncplugin import AbstractSyncPlugin
 from kitovu.sync.settings import Settings, ConnectionSettings
 from kitovu.sync.plugin import smb
 
 
-def _find_plugin(pluginname: str) -> syncplugin.AbstractSyncPlugin:
+def _find_plugin(plugin_settings: ConnectionSettings,
+                 validator: typing.Optional[utils.SchemaValidator] = None) -> AbstractSyncPlugin:
```

Comment:
Removing that from the `_find_plugin` would lead to a bit of duplicated code. And additionally in the FileCache it shouldn't create new plugin instances (or reset anything). I would suggest we provide a dicitonary of available plugins to the FileCache. For example if a plugin gets removed from the settings the FileCache shouldn't try to load it.